### PR TITLE
feat(wasm-dpp): implement identity update transition

### PR DESCRIPTION
**Pull request by markin-io**
## Issue being fixed or feature implemented

PR implements JS bindings for IdentityUpdateTransition

## What was done?

Implemented relevant bindings and adapted JS tests for them

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-dpp/src/data_trigger/dashpay_data_triggers/mod.rs [hunk 1]
```diff
@@ -41,13 +42,11 @@ where
...
 
-    let latest_block_header: JsonValue = context
+    let core_chain_locked_height = context
         .state_repository
-        .fetch_latest_platform_block_header()
-        .await?;
-
-    let core_chain_locked_height =
-        latest_block_header.get_i64(PROPERTY_CORE_CHAIN_LOCKED_HEIGHT)?;
+        .fetch_latest_platform_core_chain_locked_height()
```
Comment (markin-io) [inline packages/rs-dpp/src/data_trigger/dashpay_data_triggers/mod.rs:47]: @antouhou I assume `fetch_latest_platform_block_header` is misused here. Replaced with `fetch_latest_platform_core_chain_locked_height`

File: packages/rs-dpp/src/data_trigger/feature_flags_data_triggers/mod.rs [hunk 2]
```diff
@@ -41,15 +43,15 @@ where
         )
     })?;
 
-    let latest_block_header: JsonValue = context
+    let core_chain_locked_height = context
         .state_repository
-        .fetch_latest_platform_block_header()
-        .await?;
+        .fetch_latest_platform_core_chain_locked_height()
```
Comment (markin-io) [inline packages/rs-dpp/src/data_trigger/feature_flags_data_triggers/mod.rs:48]: @antouhou please check whether this substitution is acceptable

File: packages/rs-dpp/src/identity/state_transition/identity_update_transition/validate_identity_update_transition_state.rs [hunk 3]
```diff
@@ -147,6 +151,12 @@ where
         if !state_transition.get_public_keys_to_add().is_empty() {
             identity.add_public_keys(state_transition.get_public_keys_to_add().iter().cloned());
 
+            raw_public_keys = identity
```
Comment (markin-io) [inline packages/rs-dpp/src/identity/state_transition/identity_update_transition/validate_identity_update_transition_state.rs:154]: @antouhou fixed bug with missing identity public keys in the resulting vector that is about to be validated

File: packages/rs-dpp/src/identity/state_transition/identity_update_transition/validate_public_keys.rs [hunk 4]
```diff
@@ -55,6 +55,7 @@ pub fn validate_public_keys(
     if !duplicated_ids.is_empty() {
         validation_result
             .add_error(StateError::DuplicatedIdentityPublicKeyIdError { duplicated_ids });
+        return Ok(validation_result);
```
Comment (markin-io) [inline packages/rs-dpp/src/identity/state_transition/identity_update_transition/validate_public_keys.rs:58]: @antouhou interrupting here immediately

File: packages/wasm-dpp/src/errors/consensus/state/identity/identity_public_key_is_read_only_error.rs [hunk 5]
```diff
@@ -19,6 +19,11 @@ impl IdentityPublicKeyIsReadOnlyErrorWasm {
     pub fn get_code(&self) -> u32 {
         self.code
     }
+
+    #[wasm_bindgen(js_name=getPublicKeyIndex)]
+    pub fn public_key_index(&self) -> u32 {
```
Comment (markin-io) [inline packages/wasm-dpp/src/errors/consensus/state/identity/identity_public_key_is_read_only_error.rs:24]: Added missing method which returns key_id inernally (Same as in JS version)

File: packages/rs-dpp/src/data_trigger/dashpay_data_triggers/mod.rs (hunk 1 above)
Comment (antouhou) [reply packages/rs-dpp/src/data_trigger/dashpay_data_triggers/mod.rs:45]: Yep, that seems to be the case

File: packages/rs-dpp/src/data_trigger/feature_flags_data_triggers/mod.rs (hunk 2 above)
Comment (antouhou) [reply packages/rs-dpp/src/data_trigger/feature_flags_data_triggers/mod.rs:45]: Yep, here too

File: packages/wasm-dpp/test/integration/identity/stateTransition/IdentityUpdateTransition/validation/state/validatePublicKeysState.spec.js [hunk 6]
```diff
@@ -1,43 +1,64 @@
...
-  it('should return invalid result if there are duplicate key ids', () => {
+  it('should return invalid result if there are duplicate key ids', async () => {
     rawPublicKeys[1].id = rawPublicKeys[0].id;
 
     const result = validatePublicKeys(rawPublicKeys);
 
-    expectValidationError(result, DuplicatedIdentityPublicKeyIdError);
+    await expectValidationError(result, DuplicatedIdentityPublicKeyIdError);
 
     const [error] = result.getErrors();
```
Comment (markin-io) [inline packages/wasm-dpp/test/integration/identity/stateTransition/IdentityUpdateTransition/validation/state/validatePublicKeysState.spec.js:39]: @antouhou please check below

**Review by antouhou — APPROVED** [review]
👍

**Pull request merged**